projects/gloo/cli/pkg/cmd/gatewayapi/convert/command.go

```diff
@@ -77,18 +79,31 @@ func run(opts *Options) error {
 		return err
 	}
 
-	filesMetrics.Add(float64(len(foundFiles)))
-
 	output := NewGatewayAPIOutput()
-	var isSnapshotFile bool
-	if opts.GlooSnapshotFile != "" {
-		isSnapshotFile = true
+	var inputSnapshot *snapshot.Instance
+	snapshotFile := opts.GlooSnapshotFile
+
+	// the snapshot file comes from control plane
+	if opts.ControlPlaneName != "" {
+		snapshotFile = foundFiles[0]
 	}
 
-	if err := output.Load(foundFiles, isSnapshotFile); err != nil {
-		return err
+	if snapshotFile != "" {
+		inputSnapshot, err = snapshot.FromGlooSnapshot(snapshotFile)
```

Comment:
I really like this part! I think it's huge that we can run this command against an existing artifact. To me that feels liek the more powerful usecase

Reply:
i plan to also add a command to download the snapshot from gloo for convenience